### PR TITLE
feat: Add short sha to VersionContext Result

### LINKED ISSUE #76: Return short sha value in the result object
**Is your feature request related to a problem? Please describe.**
Currently the result object contains the full sha of the commit.  For uses to get the shortened version they need to manipulate the value themselves.

**Describe the solution you'd like**
The result object should provide the short sha (7 characters) so users do not need to manipulate any values themselves.

### PR DESCRIPTION
**Pull request by asci13**
# Description
Added Sha7 (instead of ShortSha - afaik thats one of the official names of it) to the Result object. 
Let me know if you'd prefer ShortSha instead.

Fixes #76

## Change Type

- [ ] Documentation: Changes to docs are included
- [ ] Infrastructure: Changes to build scripts/non-deliverables
- [ ] Bug Fix: Fixes an issue in implementation
- [x] New Feature: Adds new functionality in a non-breaking manner
- [ ] Breaking Change: Implements a fix or feature in a breaking manner

# Quality
- [x] Unit tests have been added/updated
- [x] Local builds/testing are successful
- [ ] Code coverage has not decreased
- [ ] Code comments have been provided (where appropriate)
- [ ] Core styles have been followed
- [ ] Documentation has been added/updated

### CODE REVIEW

**Comment by asci13** [comment]
@Kieranties as promised :) Please take a look (especially the property naming) and let me know what to change.

**Comment by Kieranties** [comment]
@asci13 Seems to be a 'quirk' with Azure Devops that the PR does not have access to to the artifacts to download simpleversion. Seems the build is ran as _you_ as the user, and you don;t have permissions to the internal feed.  I assumed the build ran as the build service user.  Will address this shortly.

File: src/SimpleVersion.Core/Pipeline/VersionContext.cs [hunk 1]
```diff
@@ -35,13 +35,15 @@ public VersionContext(Git.IRepository repository)
...
+        {
+            var sha = Repository.Head.Tip?.Sha;
             return new VersionResult
             {
                 BranchName = Repository.Head.FriendlyName,
                 CanonicalBranchName = Repository.Head.CanonicalName,
-                Sha = Repository.Head.Tip?.Sha
-            };
+                Sha = sha,
+                Sha7 = (string.IsNullOrEmpty(sha) || sha.Length < 8) ? null : sha.Substring(0, 7)
```
Comment (Kieranties) [inline src/SimpleVersion.Core/Pipeline/VersionContext.cs]: @asci13 Should this be `sha.length < 7`? If the sha is 7 or greater you should still set the `sha7`

**Comment by Kieranties** [comment]
@asci13 I've updated the build to pull simpleversion from nuget for builds.  However, Azure devops doesn't appear to be using the change for the PR.  You'll need to rebase your work on master and update the PR :-(

File: src/SimpleVersion.Core/Pipeline/VersionContext.cs (hunk 1 above)
Comment (asci13) [reply src/SimpleVersion.Core/Pipeline/VersionContext.cs]: @Kieranties yup, you're right - adjusted accordingly

**Comment by asci13** [comment]
@Kieranties all done now i hope :)

**Comment by Kieranties** [comment]
@asci13 Shipped.  You'll find the assets available on nuget.org soon.  In the mean time: https://github.com/Kieranties/SimpleVersion/releases/tag/v0.2.0-alpha1.25

**Comment by asci13** [comment]
@Kieranties no stress with the artifacts, just wanted to contribute :)

**Pull request merged**